src/lib.rs

```diff
@@ -42,456 +42,417 @@
+        if let Some(format) = captures.name("fmt") {
+            return format_to_ansi(format.as_str());
+        }
+
+        let bg = captures.name("bg").is_some();
+
+        let templater = if bg {
+            fmt_background_template
         } else {
-            FORMAT_TEMPLATES
+            fmt_template
         };
 
-        if code.len() == 6 {
+        if let Some(hex) = captures.name("hex") {
+            return hex_to_ansi(hex.as_str(), templater);
+        }
+
+        // if it's not a formatter or hex code, it's a color code
+        let color = &captures["color"];
+
+        if self.depth == Depth::High {
             let [r, g, b] =
-                [0, 2, 4].map(|i| u8::from_str_radix(&code[i..i + 2], 16).unwrap().to_string());
+                COLORS_24BIT(color).expect("the regex should match only valid color codes");
 
-            Some(fill_rgb_template(formats[&24u8], &r, &g, &b))
-        } else if let Some(value) = FORMATTERS.get(code) {
-            Some(fill_template(formats[&3u8], value))
-        } else {
-            let template = formats[&(self.depth as u8)];
+            return fill_rgb_template(templater(Depth::High), r, g, b);
+        }
 
-            if self.depth == Depth::High {
-                let [r, g, b] = COLORS_24BIT.get(code)?;
+        let color_map =
+            colors(self.depth).expect("at this point depth should only be TTY, Low or Medium");
 
-                return Some(fill_rgb_template(template, r, g, b));
-            }
+        let mapped = color_map(color).expect("the regex should match only valid color codes");
 
-            let color_map = COLORS[&(self.depth as u8)];
-            let mut value = color_map.get(code)?.to_string();
+        // low bit depths use different way of specifying background
+        let value = if bg && self.depth <= Depth::Low {
+            (mapped
+                .parse::<u8>()
+                .expect("color tables should contain valid numbers")
+                + 10)
+                .to_string()
+        } else {
+            mapped.to_string()
+        };
 
-            if bg && self.depth <= Depth::Low {
-                value = (value.parse::<u8>().ok()? + 10).to_string()
-            };
+        fill_template(templater(self.depth), &value)
+    }
 
-            Some(fill_template(template, &value))
-        }
+    /// Writes the prompt to stdout, then reads a line from input,
+    /// and returns it (excluding the trailing newline).
+    pub fn input(&self, prompt: &str) -> std::io::Result<String> {
+        print!("{}", self.convert(prompt));
+        stdout().flush()?;
+
+        let mut inp = String::new();
+        stdin().read_line(&mut inp)?;
+        Ok(inp.trim_end().to_owned())
     }
 
     /// Resets the formatting back to the default.
-    pub fn reset(&self) {
-        print!("{}", self.convert(format!("{}r", self.marker)));
+    pub fn reset() {
+        print!("\x1b[0m");
+    }
```

Comment:
Debating whether this is useful or not, it got cut from Dahlia.py v3

Reply:
idk, I‌ just left it as it was, but if you removed it, I'll prolly do it as well here for consistency